src/ext_manifest.c

```diff
@@ -131,6 +131,7 @@ static int ext_man_build(const struct module *module,
 
 out:
 	free(sec_buffer);
+	/* cppcheck-suppress memleak */
```

Comment:
I don't think that's correct, I think cppcheck found a real bug. I believe a `free(buffer)` is missing here.

Now I'm not too worried about memory leaks in a tool like `rimage` but if it's possible to fix warnings by fixing bugs then we might as well.

Reply:
I think it's complaining about the `calloc()` at line 118. `buffer` is not being freed here, hence the warning. I wonder if you can tell cppcheck to ignore this particular variable instead of ignoring all memleak. Or maybe the calling of this functions never frees `dst_buff`?

Reply:
Sorry I didn't read the whole function and missed `dst_buffer`.

You could also get rid of `buffer`, it's not very useful. I tested this and it works:
```diff
 	/* fill buffer with ext_man and section content */
-	memcpy(buffer, &ext_man, ext_man.header_size);
 	offset = ext_man.header_size;
+	memcpy(*dst_buff, &ext_man, offset);
 
-	memcpy(&buffer[offset], sec_buffer, section->size);
-
-	*dst_buff = (struct ext_man_header *)buffer;
+	memcpy((char *)*dst_buff + offset, sec_buffer, section->size);
 
 out:
 	free(sec_buffer);
```

Reply:
I rather refactor a little to silence the false positive than put down a suppressor. I will try this out. I also filed a bug upstream against this false positive

Reply:
@dcpleung i checked the caller and it handles it fine, cppcheck just fails to understand that dst_buff passes the memory out of scope and therefore is not a leak